File: spring-reactive/demo/src/main/java/com/example/demo/Controller.java
package com.example.demo;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequiredArgsConstructor
public class Controller {
    private final SlowService slowService;

    @GetMapping("/fast/{value}")
    public Mono<String> fastGet(@PathVariable("value") long value) {
        return Mono.fromCallable( () -> String.format("fast %d", value));
    }

    @GetMapping("/slow/{value}")
    public Mono<String> slowGet(@PathVariable("value") long value) {
        return Mono.fromCallable( () -> slowService.getString(value));
    }
}

File: spring-cloud/spring-cloud-gateway/books/src/main/kotlin/com/dz/books/Controller.kt
package com.dz.books

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class Controller {

    @GetMapping("")
    fun books(): List<Book> {
        return listOf(Book("book1"), Book("book2"))
    }
}
